packages/service-library/src/data-providers/page-provider.ts

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { client, CosmosContainerClient, cosmosContainerClientTypes, CosmosOperationResponse } from 'azure-services';
+import { inject, injectable } from 'inversify';
+import { Page } from 'storage-documents';
+import { GuidGenerator } from 'common';
+import _ from 'lodash';
+import { PartitionKeyFactory } from '../factories/partition-key-factory';
+
+@injectable()
+export class PageProvider {
+    public constructor(
+        @inject(cosmosContainerClientTypes.websiteRepoContainerClient) private readonly cosmosContainerClient: CosmosContainerClient,
+        @inject(GuidGenerator) private readonly guidGenerator: GuidGenerator,
+        @inject(PartitionKeyFactory) private readonly partitionKeyFactory: PartitionKeyFactory,
+    ) {}
+
+    public async createPageForWebsite(pageUrl: string, websiteId: string): Promise<void> {
+        const page = this.normalizeDbDocument({
+            id: this.guidGenerator.createGuidFromBaseGuid(websiteId),
+            websiteId: websiteId,
+            url: pageUrl,
+        });
+        await this.cosmosContainerClient.writeDocument(page);
+    }
+
+    public async updatePage(page: Partial<Page>): Promise<void> {
+        await this.cosmosContainerClient.mergeOrWriteDocument(this.normalizeDbDocument(page));
+    }
+
+    public async readPage(id: string): Promise<Page> {
+        const response = await this.cosmosContainerClient.readDocument<Page>(id, this.getPagePartitionKey(id));
+        client.ensureSuccessStatusCode(response);
+
+        return response.item;
+    }
+
+    public async readAllPagesForWebsite(websiteId: string): Promise<Partial<Page>[]> {
```

Comment:
this method may return too many results, we can consider to expose an iterable interface to the method's consumer like [generator function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function*)